src/storage/heap_file.c

```diff
@@ -20519,45 +20520,36 @@ heap_get_insert_location_with_lock (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONT
   slot_count = spage_number_of_slots (context->home_page_watcher_p->pgptr);
 
   /* find REC_DELETED_WILL_REUSE slot or add new slot */
-  /* slot_id == slot_count means add new slot */
-  for (slot_id = 0; slot_id <= slot_count; slot_id++)
-    {
-      slot_id = spage_find_free_slot (context->home_page_watcher_p->pgptr, NULL, slot_id);
-      if (slot_id == SP_ERROR)
-	{
-	  break;		/* this will not happen */
-	}
+  slot_id = spage_find_free_slot (context->home_page_watcher_p->pgptr, NULL, slot_id);
 
-      context->res_oid.slotid = slot_id;
+  context->res_oid.slotid = slot_id;
 
-      if (lock == NULL_LOCK)
-	{
-	  /* immediately return without locking it */
-	  return NO_ERROR;
-	}
+  if (lock == NULL_LOCK)
+    {
+      /* immediately return without locking it */
+      return NO_ERROR;
+    }
 
-      /* lock the object to be inserted conditionally */
-      lk_result = lock_object (thread_p, &context->res_oid, &context->class_oid, lock, LK_COND_LOCK);
-      if (lk_result == LK_GRANTED)
+  /* lock the object to be inserted conditionally */
+  lk_result = lock_object (thread_p, &context->res_oid, &context->class_oid, lock, LK_COND_LOCK);
+  if (lk_result == LK_GRANTED)
+    {
+      /* successfully locked! */
+      return NO_ERROR;
+    }
+  else if (lk_result != LK_NOTGRANTED_DUE_TIMEOUT)
```

Comment:
else if 전체를 #if !defined(NDEBUG) ~ #endif로 묶는게 좋겠네요.
```suggestion
#if !defined(NDEBUG)
  else if (lk_result != LK_NOTGRANTED_DUE_TIMEOUT)
    {
      ...
    }
#endif
```

Reply:
수정하였습니다.